ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/SourceBuilderBase.java

```diff
@@ -91,11 +92,14 @@
         planInfo
     );
 
+    final LogicalSchema stateStoreSchema = source.getSourceSchema()
+        .withPseudoColumnsToMaterialize(source.getPseudoColumnVersion());
+
     return KTableHolder.materialized(
         ktable,
         buildSchema(source, false),
         ExecutionKeyFactory.unwindowed(buildContext),
-        MaterializationInfo.builder(stateStoreName, physicalSchema.logicalSchema())
+        MaterializationInfo.builder(stateStoreName, stateStoreSchema)
```

Comment:
Thanks for this bug fix, @AlanConfluent !

Reply:
No problem.  Glad to help.